app-frontend/src/app/services/auth/auth.service.js

```diff
@@ -463,6 +463,27 @@ export default (app) => {
         fetchUserRoles() {
             return this.User.roles().$promise;
         }
+
+        isSuperOrAdmin(groupIds) {
+            let deferred = this.$q.defer();
+
+            this.$q.all({
```

Comment:
shouldn't need to do this - authService stores user roles in `this.userRoles` once they've been fetched during app init.
isSuperOrAdmin shouldn't need to be a promise, just call it directly in the tempalte and have it check the roles variable every time for the correct roles